demes/demes.py

```diff
@@ -177,12 +177,20 @@ class Epoch:
                dt = (epoch.start_time - t) / epoch.time_span
                 r = math.log(epoch.end_size / epoch.start_size)
                 N = epoch.start_size * math.exp(r * dt)
+         * ``linear``: the deme's size changes linearly from
+           ``start_size`` to ``end_size`` over the epoch.
+           If :math:`t` is a time within the span of the epoch,
+           the deme size :math:`N` at :math:`t` can be calculated as:
 
-        .. warning::
+           .. code::
+
+               dt = (epoch.start_time - t) / epoch.time_span
+                N = epoch.start_size + (epoch.end_size - epoch.start_size) * dt
+
+        .. note::
 
-            Do not assume an exponentially changing size just because
-            ``start_size != end_size``. For forwards compatibility,
-            applications should always check the ``size_function``.
+            When ``start_size != end_size`` an exponentially changing size is
+            assumed.
```

Comment:
I don't think this is the right place to mention this. The warning can probably just be removed, because now that we have "linear" as an option it should be self evident that when ``start_size != end_size`` the size function could be exponential or linear (and one must look at the `size_function` field to find out).

An exponentially changing size is only assumed when parsing a model (e.g. a from YAML, or from a demes.Builder()) and that model doesn't specify the `size_function`. But I think the docs here are mostly for people using an Epoch object, so this could potentially be confusing. A Epoch object *always* has the `size_function` set, and users should *always* check the `size_function` of the epoch to find out what to do when `start_size != end_size`.

I agree that we need to document the behaviour when reading in a model and the size_function isn't specified. But I think that should be in the spec instead.